src/v/cluster/topic_table.cc

```diff
@@ -151,6 +153,60 @@ topic_table::apply(finish_moving_partition_replicas_cmd cmd, model::offset o) {
     return ss::make_ready_future<std::error_code>(errc::success);
 }
 
+template<typename T>
+void update_if_set(std::optional<T>& property, std::optional<T> override) {
+    if (override) {
+        property = override;
+    }
+}
+
+template<typename T>
+void update_if_set(tristate<T>& property, tristate<T> override) {
+    if (override.has_value() || override.is_disabled()) {
+        property = override;
+    }
+}
+
+ss::future<std::error_code>
+topic_table::apply(update_topic_properties_cmd cmd, model::offset o) {
+    auto tp = _topics.find(cmd.key);
+    if (tp == _topics.end()) {
+        co_return make_error_code(errc::topic_not_exists);
+    }
+    auto& properties = tp->second.cfg.properties;
+    auto& overrides = cmd.value;
+    /**
+     * Update topic properties
+     */
+    update_if_set(
```

Comment:
should this be, `update_or_set` i don't follow the `update_if_set`. If a user is telling redpanda to set segment_size, then why not set it for the topic?

Reply:
When value is set i.e. optional isn't empty, we update configuration

Reply:
makes sense.